packages/core/src/model/nodes/DelayNode.ts

```diff
@@ -0,0 +1,117 @@
+import { ChartNode, NodeConnection, NodeId, NodeInputDefinition, NodeOutputDefinition, PortId } from '../NodeBase.js';
+import { nanoid } from 'nanoid';
+import { NodeImpl, NodeUIData, nodeDefinition } from '../NodeImpl.js';
+import { Inputs, Outputs } from '../GraphProcessor.js';
+import { EditorDefinition } from '../../index.js';
+import { dedent } from 'ts-dedent';
+
+export type DelayNode = ChartNode<'delay', DelayNodeData>;
+
+export type DelayNodeData = {
+  delay: number;
+};
+
+export class DelayNodeImpl extends NodeImpl<DelayNode> {
+  static create(): DelayNode {
+    const chartNode: DelayNode = {
+      type: 'delay',
+      title: 'Delay',
+      id: nanoid() as NodeId,
+      visualData: {
+        x: 0,
+        y: 0,
+        width: 150,
+      },
+      data: {
+        delay: 0,
+      },
+    };
+
+    return chartNode;
+  }
+
+  getInputDefinitions(connections: NodeConnection[]): NodeInputDefinition[] {
+    const inputs: NodeInputDefinition[] = [];
+    const inputCount = this.#getInputPortCount(connections);
+
+    for (let i = 1; i <= inputCount; i++) {
+      inputs.push({
+        dataType: 'any',
+        id: `input${i}` as PortId,
+        title: `Input ${i}`,
+      });
+    }
+
+    return inputs;
+  }
+
+  getOutputDefinitions(connections: NodeConnection[]): NodeOutputDefinition[] {
+    const outputs: NodeOutputDefinition[] = [];
+    const inputCount = this.#getInputPortCount(connections);
+
+    for (let i = 1; i <= inputCount - 1; i++) {
+      outputs.push({
+        dataType: 'any',
+        id: `output${i}` as PortId,
+        title: `Output ${i}`,
+      });
+    }
+
+    return outputs;
+  }
+
+  static getUIData(): NodeUIData {
+    return {
+      infoBoxBody: dedent`
+        Delays the execution and then passes the input value to the output without any modifications.
+      `,
+      infoBoxTitle: 'Delay Node',
+      contextMenuTitle: 'Delay',
+      group: ['Logic'],
+    };
+  }
+
+  #getInputPortCount(connections: NodeConnection[]): number {
+    const inputNodeId = this.chartNode.id;
+    const inputConnections = connections.filter(
+      (connection) => connection.inputNodeId === inputNodeId && connection.inputId.startsWith('input'),
+    );
+
+    let maxInputNumber = 0;
+    for (const connection of inputConnections) {
+      const messageNumber = parseInt(connection.inputId.replace('input', ''), 10);
+      if (messageNumber > maxInputNumber) {
+        maxInputNumber = messageNumber;
+      }
+    }
+
+    return maxInputNumber + 1;
+  }
+
+  getEditors(): EditorDefinition<DelayNode>[] {
+    return [
+      { type: 'number', label: 'Delay (ms)', dataKey: 'delay' as never, defaultValue: 0 },
```

Comment:
Oh why are you casting this to never?

Reply:
because it told me that's what it is expecting 😢 😆

Reply:
It should be typed to be the keys of the data object associated with `DelayNode`, is that all set up correctly?

Reply:
Oh can you add a `useDelayInput` too? I like to add those wherever it makes sense

Reply:
It's not complaining about it anymore, not sure what was up before.
Do you have an example of such a `useX` method? I haven't seen it before